sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamingQueryCache.scala

```diff
@@ -228,62 +212,109 @@ private[connect] class SparkConnectStreamingQueryCache(
-            logInfo(
-              log"Removing references for id: ${MDC(QUERY_ID, id)} " +
-                log"runId: ${MDC(QUERY_RUN_ID, runId)} in " +
-                log"session ${MDC(SESSION_ID, v.sessionId)} after expiry period")
-            queryCache.remove(k)
+        case Some(_) => // Inactive query waiting for expiration. Do nothing.
+          logInfo(
+            log"Waiting for the expiration for id: ${MDC(QUERY_ID, id)} " +
+              log"runId: ${MDC(QUERY_RUN_ID, runId)} in " +
+              log"session ${MDC(SESSION_ID, v.sessionId)}")
+
+        case None => // Active query, check if it is stopped. Enable timeout if it is stopped.
+          val isActive = v.query.isActive && Option(v.session.streams.get(id)).nonEmpty
 
-          case Some(_) => // Inactive query waiting for expiration. Do nothing.
+          if (!isActive) {
             logInfo(
-              log"Waiting for the expiration for id: ${MDC(QUERY_ID, id)} " +
+              log"Marking query id: ${MDC(QUERY_ID, id)} " +
                 log"runId: ${MDC(QUERY_RUN_ID, runId)} in " +
-                log"session ${MDC(SESSION_ID, v.sessionId)}")
-
-          case None => // Active query, check if it is stopped. Enable timeout if it is stopped.
-            val isActive = v.query.isActive && Option(v.session.streams.get(id)).nonEmpty
-
-            if (!isActive) {
-              logInfo(
-                log"Marking query id: ${MDC(QUERY_ID, id)} " +
-                  log"runId: ${MDC(QUERY_RUN_ID, runId)} in " +
-                  log"session ${MDC(SESSION_ID, v.sessionId)} inactive.")
-              val expiresAtMs = nowMs + stoppedQueryInactivityTimeout.toMillis
-              queryCache.put(k, v.copy(expiresAtMs = Some(expiresAtMs)))
-              // To consider: Clean up any runner registered for this query with the session holder
-              // for this session. Useful in case listener events are delayed (such delays are
-              // seen in practice, especially when users have heavy processing inside listeners).
-              // Currently such workers would be cleaned up when the connect session expires.
-            }
-        }
+                log"session ${MDC(SESSION_ID, v.sessionId)} inactive.")
+            val expiresAtMs = nowMs + stoppedQueryInactivityTimeout.toMillis
+            queryCache.put(k, v.copy(expiresAtMs = Some(expiresAtMs)))
+            // To consider: Clean up any runner registered for this query with the session holder
+            // for this session. Useful in case listener events are delayed (such delays are
+            // seen in practice, especially when users have heavy processing inside listeners).
+            // Currently such workers would be cleaned up when the connect session expires.
+          }
       }
+    })
 
-      taggedQueries.toArray.foreach { case (key, value) =>
-        value.zipWithIndex.toArray.foreach { case (queryKey, i) =>
-          if (queryCache.contains(queryKey)) {
```

Comment:
To whom it may concern - I'm pretty sure that this condition was wrong, so fixed it.

Reply:
@HyukjinKwon can you please check if the condition was right? as far as I understood, the key was supposed to be removed if no associated entry was found in queryCache. thanks!